src/cache/response.ts

```diff
@@ -56,9 +59,13 @@ export class ResponseCache<
    * @param results - the entries to write to the cache
    */
   async write(transportName: string, results: TimestampedProviderResult<T>[]): Promise<void> {
+    const censorList = CensorList.getAll()
     const entries = results.map((r) => {
+      const censoredResponse = censor(r.response, censorList) as TimestampedAdapterResponse<
```

Comment:
The censor function will not throw an error atm, and just respond with a redacted '[Unknown]'. This is not good enough for the EA responses, so we should split the censor fn into an internal one that we can use here that will float up the error so we can store a custom error response instead